Add parameter which allows users to select umap instance freely

davidbeckonline opened the pull request:
## Summary
In this PULL REQUEST (PR), I am suggesting the enhancement of the shortcode with a parameter which enables users to refer umap instances in different countries.
In it's current state, the shortcode is hard coded to domain https://umap.openstreetmap.fr. With this enhancement also https://umap.openstreetmap.de could be considered for example.

## Changes
This PR includes small changes to
* readme.md
* layouts/shortcodes/openstreetmap.html

### Default Value
In https://github.com/hanzei/hugo-component-osm/issues/7#issuecomment-3670375946 we briefly discussed the right default value for this new parameter.
I tend to think that it should be openstreetmap.fr . The intention is to not create a breaking change for users who update the submodule. Even though that might happen rarely.

## Test
Repository https://github.com/davidbeckonline/hugo-component-osm was tested in https://github.com/davidbeckonline/pegel-konstanz_website . The map from openstreetmap.de is resolving correctly on https://www.pegel-konstanz.de/02_analysis/02_02_individual/low_water_2025/index.html .

## Background
I came across this issue, when accidentally creating my map on https://umap.openstreetmap.de .
This is my first Pull Request ever. FYI.

davidbeckonline committed:
Update Readme

Update readme and elaborate on new parameter umapDomain

davidbeckonline committed:
Update html with new parameter umapDomain

Add umapDomain parameter and adjust iFrame accordingly.

davidbeckonline committed:
Correct handling of umapDomain

hanzei commented on README.md:
```diff
@@ -18,13 +18,14 @@ Now you just have to add this theme component to your themes in `config.toml`, e
 ## Creating and including a map
 
-First create a map for free on https://umap.openstreetmap.fr/en/. Then include this map by using the `openstreetmap` shortcode, e.g. `{{< openstreetmap mapName="demo-map_1" >}}`
+First create a map for free on umap. Previous versions of this repository only supported https://umap.openstreetmap.fr/en/. With the introduction of parameter `umapDomain`, the domain of umap can be adjusted. After the creation, you include this map by using the openstreetmap shortcode, e.g. `{{< openstreetmap mapName="demo-map_1" >}}` .
```
The README should (mostly) document the current state. Let's remove the comment about the pervious version.

davidbeckonline replied:
Fair point. Adjusted accordingly.

hanzei commented on layouts/shortcodes/openstreetmap.html:
```diff
@@ -22,4 +23,4 @@
 {{ $class := .Get "class" }}
 
-<iframe {{ with $class }}class="{{ . }}"{{ end }} width="{{ $mapWidth }}" height="{{ $mapHeight }}" frameBorder="0" src="https://umap.openstreetmap.fr/en/map/{{- $mapName -}}?scaleControl={{ $scaleControl }}&miniMap={{ $miniMap }}&scrollWheelZoom={{ $scrollWheelZoom }}&zoomControl={{ $zoomControl }}&allowEdit={{ $allowEdit }}&moreControl={{ $moreControl }}&searchControl={{ $searchControl }}&tilel ...
+<iframe {{ with $class }}class="{{ . }}"{{ end }} width="{{ $mapWidth }}" height="{{ $mapHeight }}" frameBorder="0" src="https://umap.{{ $umapDomain }}/en/map/{{- $mapName -}}?scaleControl={{ $scaleControl }}&miniMap={{ $miniMap }}&scrollWheelZoom={{ $scrollWheelZoom }}&zoomControl={{ $zoomControl }}&allowEdit={{ $allowEdit }}&moreControl={{ $moreControl }}&searchControl={{ $searchControl }}&tile ...
```
Are we sure that every URL starts with `umap`? I wonder if we should definde the full URL via the config e.g. `https://umap.openstreetmap.fr`

davidbeckonline replied:
Starting from https://umap-project.org/, there are three options offered, when clicking on "create map":
* https://umap.openstreetmap.de
* https://umap.osm.ch
* https://umap.openstreetmap.fr

Still, it might be a good idea to define this in a more generic way. Adjusted accordingly.

hanzei reviewed (changes requested):
Thanks!

davidbeckonline committed:
Update readme.md

Incorporate feedback from https://github.com/hanzei/hugo-component-osm/pull/8

davidbeckonline committed:
Update openstreetmap.html with full domain

Updated parameter umapDomain so that the entire domain / Url is considered. See discussion in https://github.com/hanzei/hugo-component-osm/pull/8

davidbeckonline commented:
Updated for review.

PS: I tried to squash my commits, but that did not work properly. I will leave it with individual commits for now. FYI.

hanzei commented on layouts/shortcodes/openstreetmap.html:
```diff
@@ -1,3 +1,4 @@
+{{ $umapDomain := .Get "umapDomain" | default "https://www.openstreetmap.fr"}}
```
Nit pick: The parameter should be called `umapURL`. It's not a domain.

davidbeckonline replied:
Agreed. Thank you for pointing this out.

Updated to `umapURL` in the html and readme file.

The updated code is running successfully in 
https://www.pegel-konstanz.de/02_analysis/02_02_individual/low_water_2025/index.html

hanzei reviewed (changes requested):
One nit pick, then we are good to go.

davidbeckonline committed:
Update umap parameter from 'umapDomain' to 'umapURL'

davidbeckonline committed:
Change umapDomain to umapURL in OpenStreetMap shortcode

hanzei reviewed (approved):
Thanks @davidbeckonline 🚀
